tests/Microsoft.Identity.Test.Unit/ExceptionTests/MsalExceptionTests.cs

```diff
@@ -509,6 +509,59 @@ await app.AcquireTokenForClient(TestConstants.s_scope)
             }
         }
 
+        [TestMethod]
+        public async Task AuthorityInNotFoundExceptions()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                harness.HttpManager.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.AuthorityCommonTenant + TestConstants.DiscoveryEndPoint));
+                harness.HttpManager.AddMockHandlerContentNotFound(HttpMethod.Post);
+
+                ConfidentialClientApplication app = null;
+                var certificate = new X509Certificate2(
+                    ResourceHelper.GetTestResourceRelativePath("RSATestCertDotNet.pfx"));
+
+                app = ConfidentialClientApplicationBuilder
+                    .Create(TestConstants.ClientId)
+                    .WithAuthority(new System.Uri(ClientApplicationBase.DefaultAuthority), true)
+                    .WithRedirectUri(TestConstants.RedirectUri)

```

Comment:
If you add a region here, will the authority in the ex be region.login.microsoft.com ? That's the scenario we try to debug really.

Reply:
The region is not preserved in the `Authority` object, however, it is available to add to the message.

Reply:
See latest commit